token/program-2022/src/instruction.rs

```diff
@@ -409,6 +431,148 @@ pub enum TokenInstruction {
+    /// `u64`.
+    ///
+    /// Return data can be fetched using `sol_get_return_data` and deserializing
+    /// the return data as a little-endian `u64`.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[]` The mint to calculate for
+    GetAccountLength,
+    /// Initializes the empty mixin on a new account.
+    ///
+    /// The account must have a size 356, which requires it to use this mixin.
+    /// The size of 356 comes from the size of a multisig (355) plus 1 byte of
+    /// padding.
+    ///
+    /// Fails if the account has already been initialized, so must be called before
+    /// `InitializeAccount`. The account must have enough data allocated for all
+    /// mixins included in the mint.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[writable]`  The account to initialize.
+    InitializeEmptyMixin,
+    /// Initialize the close account mixin on a new mint.
+    ///
+    /// Fails if the mint has already been initialized, so must be called before
+    /// `InitializeMint`.
+    ///
+    /// The mint must have exactly enough space allocated for the base mint (82
+    /// bytes), plus 83 bytes of padding, 1 byte reserved for the account type,
+    /// then space required for the mixin.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[writable]` The mint to initialize.
+    InitializeMintCloseMixin,
+    /// Initialize the transfer fee mixin on a new mint.
+    ///
+    /// Fails if the mint has already been initialized, so must be called before
+    /// `InitializeMint`.
+    ///
+    /// The mint must have exactly enough space allocated for the base mint (82
+    /// bytes), plus 83 bytes of padding, 1 byte reserved for the account type,
+    /// then space required for the mixin.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[writable]` The mint to initialize.
+    InitializeMintTransferFeeMixin {
+        /// Pubkey that may update the fees
+        fee_config_authority: COption<Pubkey>,
+        /// Harvest instructions must be signed by this key
+        fee_withdraw_authority: COption<Pubkey>,
+        /// Amount of transfer collected as fees, expressed as basis points of the
+        /// transfer amount
+        transfer_fee_basis_points: u16,
+        /// Maximum fee assessed on transfers
+        maximum_fee: u64,
+    },
+    /// Initializes the transfer fee mixin on a new account.
+    ///
+    /// Fails if the account has already been initialized, so must be called before
+    /// `InitializeAccount`. The account must have enough data allocated for all
+    /// mixins included in the mint.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[writable]`  The account to initialize.
+    ///   1. `[]` The mint this account will be associated with.
+    InitializeAccountTransferFeeMixin,
```

Comment:
Ah ok, but then for the other ones, ie. ZKToken, that will have an explicit initializer, correct?

Reply:
Yes.  In the ZkToken case it's likely that this'll often happen after account initialization though.  Specifically:
1. ATA program creates an account for the wallet via some on-ramp
2. User's wallet later initializes the ZkToken mixin and provides the user's ElGamal pubkey. `InitializeZkToken` internally performs a realloc to make room for the ZkToken mixin, since the ATA program wants to pay as little as possible to initialize the user's account.

Reply:
Sweet, sounds good to me